src/vmm/src/device_manager/mmio.rs

```diff
@@ -360,6 +361,32 @@ impl MMIODeviceManager {
         Ok(())
     }
 
+    /// Run fn for each registered device.
+    pub fn for_each_virtio_device<F, E>(&self, mut f: F) -> std::result::Result<(), E>
```

Comment:
You can use this newly added specialized fn to simplify code in `fn kick_devices()` below and possibly even `fn with_virtio_device_with_id()`

Reply:
Nice. I used it in `kick_devices()`. `with_virtio_device_with_id` uses `get_device` which is more efficient.